controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -384,9 +387,28 @@ public CompletableFuture<CreateStreamStatus> createStream(String scope, String s
             return CompletableFuture.completedFuture(
                     CreateStreamStatus.newBuilder().setStatus(CreateStreamStatus.Status.INVALID_STREAM_NAME).build());
         }
+        try {
+            validateStreamConfig(streamConfig);
+        } catch (IllegalArgumentException e) {
```

Comment:
This appears to be using IllegalArugmentException as flow control. As this is not an unexpected case it should be handled by a return value.